src/PropertyList/PropertyList.tsx

```diff
@@ -17,36 +16,38 @@ export interface Property {
 
 interface BaseProps {
     /** This is to set the list of the property- caption, captionStyle, content and contentStyles */
-    attributes: Immutable.List<Property>;
+    attributes: Property[];
     /** This is to set the external class on the Property List. */
     className?: string;
     /** This is to set the data for the content defined in the Property */
     data: any;
 }
 
 export interface PropertyListProps extends BaseProps {
-    /** This is to set the alignment of the content. Set variant as 'left-align' for aligning the content to left. By default it is right aligned.*/
+    /** This is to set the caption width. Set variant as 'variable' to render the caption according to the size of the caption.
+     *  The default is set to Fixed. */
     variant: string;
 }
 
-/** The PropertyList Component is used to display the list of captions and their corresponding data. The default behavior of the content is right align but you can choose to
- *  align it to left by setting variant='left-align'. The custom styles can be applied to the caption and the corresponding content.
+/** The PropertyList Component is used to display the list of captions and their corresponding data. The default behavior of the caption is fixed but you can choose to
+ *  set the variant='Variable'. The custom styles can be applied to the caption and the corresponding content.
+ *  For more information please check the implementations below.
  */
 const PropertyList: React.FC<PropertyListProps> = (props) => {
     const { attributes, data, variant } = props;
 
-    if (variant && variant === 'left-align') {
-        return <PropertyListLeftAligned {...props} />;
+    if (variant && variant === 'variable') {
+        return <PropertyListVariable {...props} />;
     }
 
     return (
         <ThemeProvider theme={theme}>
             <AttributeGrid>
                 {attributes.map(a => (
-                    <React.Fragment key={a?.caption}>
```

Comment:
Aaah, sorry, I steered you wrong.  We do need the full `React.Fragment` syntax here in order to specify the key.